### ISSUE TITLE
Null pointer instance

### ISSUE THREAD

**Issue by youarebamboozled** [CallMeMax/enfusion-REST-sql#1]
NULL pointer to instance. Variable 'm_sqlQuery' during call 'L1960_SQLQuery.Count'
when i do this : 
auto m_sqlQuery = L1960_SQL.Query(0,"SELECT CURRENT_TIMESTAMP() as time");
		foreach (auto ele : m_sqlQuery)
		{
		  Print(ele.GetString("Name"));
			Print("test");
		}
why?

**Comment by CallMeMax** [comment]
Hi,
so there are two possibilities why `L1960_SQL.Query()` returnes null. First because the config file contains an invalid server name. In this case, there should be log message saying: 
`RestSQL Error: config file does not contain valid server name <confg.file>`.
Second the REST Api itself might return null. This can have multiple reasons and I can only guess without access/knowledge of your setup - you need to debug that yourself. Maybe there is no `restAPI.php` on the server or it cannot be reached? I suggesting setting a break point where you perform the Query and step through the code.

**Comment by youarebamboozled** [comment]
Theres also the problem that L1960_SQLREST.Query(0,"SELECT CURRENT_TIMESTAMP() as time") is basically wrong because theres no L1960_SQLREST class. What about that? And I also get always the Error SCRIPT    (F): RestSQL Error: config file does not contain valid server name ``
SCRIPT    (E): async query failed!. Eventhough I created a (i think valid json, which btw isnt being created automatically)

**Comment by CallMeMax** [comment]
Yes, I removed `L1960_SQLREST` from the README, but in the code this class is never mentioned so the example in the `GameComponent` showed the correct behavior the entire time.

The script error seems to point at your problem. As I already suspected in my first respons, the problem could be that you did not define a valid config file. Next time I suggest you share all information from the start - that helps me to give better answers.

The config file is created using the JSON api from Arma itself. If you are not familiar how it works, I suggest looking at the examples and the code. This issue here is not the right place for general Arma coding advice.

As the script error tells you, the problem is definitely because your config file could not be read. A simple solution could be to delete the config, start the workbench project, let arma create a new config, put in the correct data and restart the workbench.
